flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -339,12 +338,26 @@ private static Request createRequest(String targetAddress, String targetUrl, Htt
 			.thenComposeAsync(
 				channel -> {
 					ClientHandler handler = channel.pipeline().get(ClientHandler.class);
-					CompletableFuture<JsonResponse> future = handler.getJsonFuture();
+
+					CompletableFuture<JsonResponse> future;
+					boolean success = false;
+
 					try {
-						httpRequest.writeTo(channel);
+						if (handler == null) {
+							throw new IOException("Netty pipeline was not properly initialized.");
+						} else {
+							httpRequest.writeTo(channel);
+							future = handler.getJsonFuture();
+							success = true;
+						}
 					} catch (IOException e) {
```

Comment:
We should catch all exceptions here, then this issue wouldn't have occurred in the first place.

Reply:
We retry the send operation if we see an `IOException`. If any other exception occurs (e.g. `RuntimeException`), this might indicate that there is another problem where we don't want to retry. Therefore, I think it is ok to only catch the `IOException`. The request will also be failed if we see a different exception.